daemon/cluster/services.go

```diff
@@ -111,7 +111,8 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string) (*apity
 			digestImage, err := c.imageWithDigestString(ctx, ctnr.Image, authConfig)
 			if err != nil {
 				logrus.Warnf("unable to pin image %s to digest: %s", ctnr.Image, err.Error())
-				resp.Warnings = append(resp.Warnings, fmt.Sprintf("unable to pin image %s to digest: %s", ctnr.Image, err.Error()))
+				// warning in the client response should be concise
+				resp.Warnings = append(resp.Warnings, fmt.Sprintf("image %s could not be accessed on a registry to record its digest. Each node will access %s independently, possibly leading to different nodes running different versions of the image.", ctnr.Image, ctnr.Image))
```

Comment:
Defined a function.

While not pretty, I think hardcoded formatting might be good enough here, especially since we are formatting a pre-defined string, more or less. If required, it's possible to add some logic based on length of the image, all of which will still be contained in one function.